### DISCUSSION TITLE
Just moved from BW and struggling to get TOTP working via camera

### DISCUSSION THREAD

**Discussion by thorrrr** [dani-garcia/vaultwarden#2896] [Q&A]
Just moved from BW and struggling to get TOTP working via camera. I tried scanning the q code but VW would not recognise it.
If i try with any other 2fa app it scans instantly!! Does this mean VW does not support the scanning of 2fa codes or is this a bug?

**Comment by BlackDex** [comment]
Could you give a bit more details?
Which totp qr?? Since the web-vault is just the same web-vault as bitwarden.

Maybe also provide some screenshots or something.

**Comment by thorrrr** [comment]
HI
I dont think it would be safe for me to send you my qcode for my 2fa account
I can blank out part of it if that is helpful

**Reply by BlackDex** [reply]
Well, if you say the qr is not valid that would be strange.
Also, you can still remove it and create a new one afterwards.

**Comment by thorrrr** [comment]
As mentioned if i use Google or Authy works instantly but BW app nope

**Reply by BlackDex** [reply]
If you mean the Bitwarden mobile client, then we can't help. Those clients are not maintained by us.

**Reply by thorrrr** [reply]
So the only way to get TOTP working with VW is the enter them manually ?

**Reply by BlackDex** [reply]
This has nothing todo with Vaultwarden it self.

If you say that the Bitwarden mobile client is unable to scan the QR, that is a mobile client issue, which, again, are not maintained by Vaultwarden. I suggest to report this at Bitwarden. But for me scanning a QR with the mobile client works just fine.

**Comment by thorrrr** [comment]
Cheers for trying i would have hoped my Phone 14 Pro would have had a good enough camera  :)
